admin_solutions/authentication.adoc

```diff
@@ -0,0 +1,166 @@
+master configuration file.
+
+[[configuring-auth-on-master]]
+=== Configuring Authentication on the Master
+
+. If you have:
+.. Already completed installation of Openshift, then copy the
+*_master-config.yaml_* file into a new directory; for example:
++
+----
+$ mkdir githubconfig; cp master-config.yaml githubconfig
+----
+.. Not yet installed OpenShift, then start the OpenShift API server, specifying
+the hostname of the (future) OpenShift master and a directory to store the
+configuration file created by the start command:
++
+----
+$ openshift start master --public-master=<apiserver> --write-config=<directory>
+----
++
+For example:
++
+----
+$ openshift start master --public-master=https://myapiserver.com:8443 --write-config=githubconfig
+----
++
+[NOTE]
+====
+If you are installing by Ansible, then you must provide Ansible with the
+identityProvider configuration. When installing by Ansible, you probably do not
+want to be manually modifying your configuration, because you would lose the
+modifications when you re-ran your install, or upgraded. This topic shows you
+how to manually handle your configuration, but Ansible users should see ????.
+
+NeedInfo: there is a place you can plug in the identity provider config and
+Ansible will drop it into your config for you (see Jason or Scott for details).
+====
++
+[NOTE]
+====
+Using `openshift start master` on its own would auto-detect hostnames, but
+GitHub must be able to redirect to the exact host name that you specified when
+registering the application. For this reason, you cannot auto-detect the ID
+because it might redirect to the wrong address. Instead, you must specify the
+hostname that web browsers use to interact with your OpenShift cluster.
+====
+. Edit the new *_master-config.yaml_* file's `*identityProviders*` stanza.
+. Copy
+link:../install_config/configuring_authentication.html#GitHub[the example `*GitHubIdentityProvider*` configuration]
+and paste it to replace the existing stanza.
+. Make the following modifications to the `*identityProviders*` stanza:
+.. Change the provider `*name*` to match the callback URL you configured on
+GitHub.
++
+For example, if you defined the callback URL as
+`https://myapiserver.com:8443/oauth2callback/github/` then the `*name*` must be
+`github`.
+.. Change the `*clientID*` to the Client ID from GitHub that you
+link:../admin_solutions/authentication.html#register-app-on-github[registered previously].
+.. Change the `*clientSecret*` to the Client Secret from GitHub that you
+link:../admin_solutions/authentication.html#register-app-on-github[registered previously].
+.. Change the `*organizations*` to include a list of one or more GitHub
+organizations to which a user must have membership in order to authenticate. If
+specified, only GitHub users that are members of at least one of the listed
+organizations will be allowed to log in. If this is not specified, then any
+person with a valid GitHub account can log in.
```

Comment:
There's some "Change the X to include...". These should probably be "Change X to include..." or "Change the X value to include...".